src/pages/home/library/Mint.tsx

```diff
@@ -0,0 +1,451 @@
+import {
+  Box,
+  FormControlLabel,
+  Button as MUIButton,
+  Stack,
+} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import { useLocation, useNavigate } from "react-router";
+import { useWindowDimensions } from "common";
+import {
+  Alert,
+  Button,
+  Checkbox,
+  HorizontalLine,
+  Switch,
+  TextInput,
+  Typography,
+} from "elements";
+import theme from "theme";
+import { Song } from "modules/song";
+import { ChangeEvent, useEffect, useState } from "react";
+import AddOwnerModal from "../uploadSong/AddOwnerModal";
+
+interface FutureOwner {
+  email: string;
+  firstName: string;
+  isCreator: boolean;
+  isRightsOwner: boolean;
+  lastName: string;
+  percentage: number;
+  role: string;
+}
+
+const Mint = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const windowWidth = useWindowDimensions()?.width;
+  const { id = "", title } = location.state as Song;
+
+  const [isExclusiveCreator, setIsExclusiveCreator] = useState(false);
+  const [hasAgreedToTerms, setHasAgreedToTerms] = useState(false);
+  const [showWarning, setShowWarning] = useState(true);
+  const [isMintSwitchOn, setIsMintSwitchOn] = useState(false);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isFirstStep, setIsFirstStep] = useState(true);
+  const [isFullyOwned, setIsFullyOwned] = useState(false);
+  const [futureOwners, setFutureOwners] = useState<FutureOwner[]>([]);
+  const [futureCreditors, setFutureCreditors] = useState<
+    Partial<FutureOwner>[]
+  >([]);
```

Comment:
The reason why i didnt add a formik around this was because i wasn't sure what all was going to be submitted as part of the owners/creditors. I figured we could add it later once we know what the structure would look like?

Reply:
given that we aren't sure what the end result would look like, we can add that later